src/components/Modal.js

```diff
@@ -71,7 +73,12 @@ export default hh(class Modal extends Component {
                 onClick: onDismiss
               }, 'Cancel') :
               null,
-            okButton
+            Utils.cond(
+              [okButton === undefined, () => buttonPrimary({ onClick: onDismiss }, 'OK')],
+              [_.isString(okButton), () => buttonPrimary({ onClick: onDismiss }, okButton)],
+              [_.isFunction(okButton), () => buttonPrimary({ onClick: okButton }, 'OK')],
+              () => okButton
+            )
```

Comment:
Discussion: I appreciate the flexibility offered here, but this seems rather overloaded. Another approach would be keeping the base component simple, and using composition to create additional components that handle specialized cases.

Reply:
+1 to @panentheos's concern. I'd love to see an example of the composition approach, perhaps in a follow-up PR?